x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportEvaluateDataFrameAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ml.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.search.SearchAction;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.action.EvaluateDataFrameAction;
+import org.elasticsearch.xpack.core.ml.dataframe.evaluation.Evaluation;
+import org.elasticsearch.xpack.core.ml.dataframe.evaluation.EvaluationResult;
+
+public class TransportEvaluateDataFrameAction extends HandledTransportAction<EvaluateDataFrameAction.Request,
+    EvaluateDataFrameAction.Response> {
+
+    private final Client client;
+
+    @Inject
+    public TransportEvaluateDataFrameAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+        super(EvaluateDataFrameAction.NAME, transportService, actionFilters, EvaluateDataFrameAction.Request::new);
+        this.client = client;
+    }
+
+    @Override
+    protected void doExecute(Task task, EvaluateDataFrameAction.Request request,
+                             ActionListener<EvaluateDataFrameAction.Response> listener) {
+        Evaluation evaluation = request.getEvaluation();
+        SearchRequest searchRequest = new SearchRequest(request.getIndices());
+        searchRequest.source(evaluation.buildSearch());
+
+        ActionListener<EvaluationResult> resultListener = ActionListener.wrap(
+            result -> listener.onResponse(new EvaluateDataFrameAction.Response(result)),
+            listener::onFailure
+        );
+
+        client.execute(SearchAction.INSTANCE, searchRequest, ActionListener.wrap(
+            searchResponse -> evaluation.evaluate(searchResponse, resultListener),
```

Comment:
I thought I had commented on this previously but the message seems to have been lost. 

`evaluation.evaluate()` should run in a different thread